### PR TITLE
feat(dpi): refactor DPI UI

### PR DESCRIPTION
**Pull request by m-dilorenzi**
This PR contains a rework of the DPI Filter UI, replacing the old card-based rule editor with a table-driven interface and introducing application groups as a first-class concept.

### Main changes

- **DPI rules**: rules are now shown in a sortable table with drag & drop reordering, inline enable/disable, and dedicated create/edit and rename drawers replacing the previous multi-purpose modal.
- **Application groups**: new tab (replacing *Exceptions*) to create, edit and delete reusable groups of applications, protocols and their categories, with a guided multi-step creation flow, search and pagination.
- **Data layer**: DPI logic moved out of components into dedicated composables (`useDpiRules`, `useApplicationGroups`, `useDpiCatalog`) built on TanStack  Query, with typed payloads, shared query keys and centralized cache invalidation. The catalog composable merges the netifyd catalog with the loaded applications/protocols and exposes categories with icons and availability info.
- **Subscription awareness**: loaded application/protocol queries are invalidated when the subscription status changes, so the catalog reflects what is actually available.
- **Shared components**: `NeMultiTextInput` gains a `helperText` prop and improved spacing; `NeStepper` gains configurable bar size and color classes.
- **Tests**: added unit tests for the three new composables.
- **i18n**: new English and Italian strings for rules and application groups.

### CODE REVIEW

File: src/stores/standalone/subscription.ts [hunk 1]
```diff
@@ -23,7 +29,14 @@ export const useSubscriptionStore = defineStore('subscription', () => {
...
       .then((res: SubscriptionStatusResponse) => {
-        isActive.value = res.data.active ?? false
+        const active = res.data.active ?? false
+
+        if (active !== isActive.value) {
+          queryClient.invalidateQueries({ queryKey: DPI_LOADED_APPLICATIONS_KEY })
+          queryClient.invalidateQueries({ queryKey: DPI_LOADED_PROTOCOLS_KEY })
+        }
+
+        isActive.value = active
```
Comment (Tbaile) [inline src/stores/standalone/subscription.ts]: We can just invalidate the queries, it's data saved in the machine, not a big issue

File: src/components/standalone/NeStepper.vue [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
+  currentStep,
+  stepLabel,
+  barSize = 'sm',
+  labelColorClasses = 'text-primary-600 dark:text-primary-500',
+  barColorClasses = 'bg-primary-600 dark:bg-primary-500'
+} = defineProps<{
   totalSteps: number
   currentStep: number
   stepLabel: string
+  barSize?: 'sm' | 'md' | 'lg' | 'xl'
```
Comment (Tbaile) [inline src/components/standalone/NeStepper.vue]: Let's keep just what we use
```suggestion
  barSize?: 'sm' | 'md'
```

Comment (Tbaile) [inline src/components/standalone/NeStepper.vue]: Why generating more than we need?

File: src/components/standalone/StandaloneTopBar.vue
Comment (Tbaile) [inline src/components/standalone/StandaloneTopBar.vue:1]: Unrelated fix, let's not change what we don't need

File: src/composables/useApplicationGroups.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+          search: params.search.value || undefined,
+          limit: params.pageSize.value,
+          page: params.page.value
+        },
+        { signal }
+      ),
+    select: (res) => res.data.values,
+    placeholderData: keepPreviousData
+  })
+}
```
Comment (Tbaile) [inline src/composables/useApplicationGroups.ts]: Not liking this separation, should be one single call with optional parameters

File: src/composables/useApplicationGroups.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+export function useCreateApplicationGroup() {
+  const invalidate = useApplicationGroupsInvalidation()
+
+  return useMutation({
+    mutationFn: (payload: DpiApplicationGroupPayload) =>
+      ubusCall<AppGroupIdResponse>('ns.dpi', 'add-appgroup', payload),
+    onSuccess: invalidate
+  })
+}
```
Comment (Tbaile) [inline src/composables/useApplicationGroups.ts]: This is a single usage function inside a separate file, should stay in the create modal

File: src/composables/useApplicationGroups.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+export function useEditApplicationGroup() {
+  const invalidate = useApplicationGroupsInvalidation()
+
+  return useMutation({
+    mutationFn: (payload: DpiApplicationGroupPayload & { id: string }) =>
+      ubusCall<AppGroupIdResponse>('ns.dpi', 'edit-appgroup', payload),
+    onSuccess: invalidate
+  })
+}
```
Comment (Tbaile) [inline src/composables/useApplicationGroups.ts]: This is a single usage function inside a separate file, should stay in the edit drawer

File: src/composables/useApplicationGroups.ts [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+export function useDeleteApplicationGroup() {
+  const invalidate = useApplicationGroupsInvalidation()
+
+  return useMutation({
+    mutationFn: (id: string) =>
+      ubusCall<{ data: { message: string } }>('ns.dpi', 'delete-appgroup', { id }),
+    onSettled: invalidate
+  })
+}
```
Comment (Tbaile) [inline src/composables/useApplicationGroups.ts]: This is a single usage function inside a separate file, should stay in the delete modal

File: src/composables/useApplicationGroups.ts
Comment (Tbaile) [inline src/composables/useApplicationGroups.ts:1]: Most of the functions here are separated for no actual reason, should stay in the Vue file that owns the data, since we're using Tanstack the data is saved into session, so we can avoid repeating it

File: src/composables/useDpiCatalog.ts
Comment (Tbaile) [inline src/composables/useDpiCatalog.ts:1]: This file has the same issue, having separate exported functions that are being used just one time

File: src/components/standalone/dpi/RenameDpiRuleDrawer.vue [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+const { t } = useI18n()
+
+const name = ref('')
+const validationBag = ref(new MessageBag())
+
+const { mutate: renameRule, isPending } = useRenameDpiRule()
+
+const schema = v.object({
+  name: v.pipe(v.string(), v.trim(), v.minLength(1, 'required'), v.maxLength(64, 'name_too_long'))
+})
```
Comment (Tbaile) [inline src/components/standalone/dpi/RenameDpiRuleDrawer.vue]: Let validation be done in the backend, single point of truth

File: src/composables/useDpiCatalog.ts [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+  return useQuery({
+    queryKey: DPI_LOADED_PROTOCOLS_KEY,
+    queryFn: ({ signal }) =>
+      ubusCall<Response<LoadedEntry[]>>('ns.dpi', 'list-loaded-protocols', {}, { signal }),
+    select: (res) => res.data.values,
+    staleTime: 5 * 60 * 1000
+  })
+}
+
+export function useDpiCatalog(kind: Ref<DpiCatalogKind>, categoryId: Ref<string>) {
```
Comment (Tbaile) [inline src/composables/useDpiCatalog.ts:171]: Why not making this into the backend? Is there a reason? A single call for all the data joined

File: src/components/standalone/NeMultiTextInput.vue [hunk 9]
```diff
@@ -51,6 +52,18 @@ const emit = defineEmits(['delete-item', 'add-item', 'update:modelValue'])
 const keys = ref<string[]>([])
 const items = ref<string[]>([])
 
+const showHelperText = computed(
+  () =>
+    !!props.helperText &&
```
Comment (Tbaile) [inline src/components/standalone/NeMultiTextInput.vue]: This is to check if the text is true or false, you want an `undefined` check

File: src/components/standalone/dpi/DeleteDpiRuleModal.vue [hunk 10]
```diff
@@ -1,99 +1,74 @@
...
+  close: []
+  deleted: []
+}>()
 
 const { t } = useI18n()
 
-const loading = ref({
-  deleteRule: false
-})
+const deleteError = ref<Error>()
```
Comment (Tbaile) [inline src/components/standalone/dpi/DeleteDpiRuleModal.vue]: this can be extracted by error inside the mutation

File: src/components/standalone/dpi/CreateApplicationGroupModal.vue [hunk 11]
```diff
@@ -0,0 +1,716 @@
...
+
+const nameSchema = v.object({
+  name: v.pipe(
+    v.string(),
+    v.trim(),
+    v.minLength(1, 'required'),
+    v.maxLength(NAME_MAX_LENGTH, 'name_too_long')
+  )
+})
+
```
Comment (Tbaile) [inline src/components/standalone/dpi/CreateApplicationGroupModal.vue]: Move this into backend

File: src/components/standalone/dpi/CreateApplicationGroupModal.vue
Comment (Tbaile) [inline src/components/standalone/dpi/CreateApplicationGroupModal.vue:1]: This approach might not be the best. While all code seem fine we are having a single component behave completely differently by it's state, probably extracting a presentation component and then two components that apply logic (create/edit) is the best approach here

File: src/components/standalone/dpi/DeleteApplicationGroupModal.vue [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+}>()
+
+const emit = defineEmits<{
+  close: []
+  deleted: []
+}>()
+
+const { t } = useI18n()
+
+const deleteError = ref<Error>()
```
Comment (Tbaile) [inline src/components/standalone/dpi/DeleteApplicationGroupModal.vue]: Error can be extracted by the mutation, if you need different behaviours you can just apply a watch function

File: src/components/standalone/dpi/CreateOrEditDpiRuleDrawer.vue
Comment (Tbaile) [inline src/components/standalone/dpi/CreateOrEditDpiRuleDrawer.vue:1]: This might have the same "issue" of the CreateApplicationModal, single component for multiple usages

**Review by Tbaile — COMMENTED** [review]
The notes I've left in the useApplicationGroups are valid for all the new composables, also there's no point to test looping functions, tests only things that eventually will break if data underneath changes or edge cases.
Validation should stick to the backend *unless needed*, the only validation I could get behind is the step-by-step one, and I'm starting to think that is better to hit a backend API for validation only.
